packages/frontend/core/src/components/affine/auth/oauth.tsx

```diff
@@ -64,9 +64,25 @@ function OAuthProvider({
   popupWindow: (url: string) => void;
 }) {
   const serverService = useService(ServerService);
+  const auth = useService(AuthService);
   const { icon } = OAuthProviderMap[provider];
 
   const onClick = useCallback(() => {
+    if (
+      scheme &&
+      (BUILD_CONFIG.isElectron || BUILD_CONFIG.isIOS || BUILD_CONFIG.isAndroid)
+    ) {
+      auth
+        .oauthPreflight(provider, scheme)
+        .then(oauthUrl => {
+          popupWindow(oauthUrl);
+        })
+        .catch(e => {
```

Comment:
you can turn this into `useAsyncCallback` to let error boundary handle error

Reply:
Really useful!

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/hTwOityLamd4hitrae7M/f0435455-3f73-4837-a969-136021fb459a.png)

Reply:
It seems I made a mistake, I need to use the notify popup myself, I'll add it

![image.png](https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/hTwOityLamd4hitrae7M/7aa016ae-a326-402a-8486-95e16cabab31.png)